notebook-lm-clone/src/memory/memory_layer.py

```
@@ -0,0 +1,326 @@
+            
+            # Save assistant response
+            self.external_memory.save(
+                rag_result.response,
+                metadata=assistant_meta
+            )
+            self._save_source_context(rag_result.sources_used)
+            
+            logger.info(f"Saved conversation turn with {len(rag_result.sources_used)} sources")
+            
+        except Exception as e:
+            logger.error(f"Error saving conversation turn: {str(e)}")
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)